apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -585,6 +482,136 @@ impl Plugin for Telemetry {
 }
 
 impl Telemetry {
+    /// This method can be used instead of `Plugin::new` to override the subscriber
+    pub async fn new_with_subscriber<S>(
```

Comment:
+1 to this question

Conversely, if this is for consumption by end-users, can a subscriber be specified when telemetry is configured through YAML?

Reply:
it is public because we must use it with test-span in the integration tests: https://github.com/apollographql/router/pull/1463/files#diff-0654669b073b6481f538035cfdc0ad46b209207837121627f534ec8c90fa1432R698